### PR TITLE
Feature/hall sensor calibration script

### PR DESCRIPTION
**Pull request by tstastny**
Should be merged AFTER #22 

Automated script for detecting calibration steps and fitting a polynomial curve to the data. Outputs int32 values for PX4 params.

Example graphical output:
screen_hall_cal

### CODE REVIEW

File: 08_analysis_tools/CalibrateHallSensor.m [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+% starting angle (prior to this, it is somewhat helpful to make some rapid
+% / large motions with the vane such that it is easier to find the
+% calibration starting point in the log data)
+% 5) hold the vane still at the starting angle for several seconds, then
+% move on directly to the next angles, pausing at each for several seconds,
+% until reaching the end angle.
+% NOTE: it is advised to take steps of 5 degrees at larger angles, until
+% reaching e.g. +/- 5-10 degrees, where the step size shold be reduced
+% (e.g. 1 or 2 degrees) in an attempt to give a higher accuracy fit at the
+% more common values the vane will take while in flight.
```
Comment (acfloria) [inline 08_analysis_tools/CalibrateHallSensor.m]: Maybe add here which params correspond to which limit, other than that the instructions are very clear

Comment (tstastny) [reply 08_analysis_tools/CalibrateHallSensor.m]: I made a small modification to this note about what calibration option to set the small range with -- were you referring to this, or something else?

File: 02_helper_functions/PrintFlightMetricsSummary.m [hunk 2]
```diff
@@ -1,27 +1,30 @@
...
+        vGnd = sqrt(sysvector.vehicle_global_position_0.vel_n.Data.^2+sysvector.vehicle_global_position_0.vel_e.Data.^2);
+        vGndMean = mean(vGnd(iLaunchGnd:iLandGnd));
+        FlightDistanceAir = vAirMean * FlightTime;
+        FlightDistanceGnd = vGndMean * FlightTime;
+        str = sprintf('***********************\n*** Flight Summary: ***\n***********************\ntLaunch: %.2fs tLand:%.2fs tFlight: %.2fs (%s dd:hh:mm:ss) dist_air: %.2fkm dist_gnd: %.2fkm\n', tLaunch, tLand, FlightTime, datestr(FlightTime/86400.0, 'DD:HH:MM:SS'),FlightDistanceAir/1000.0,FlightDistanceGnd/1000.0);
+        disp(str);
+    end
 
+else
+    disp('INFO: No flight detected.');
```
[the remark this answers is not in the document]
Comment (tstastny) [reply 02_helper_functions/PrintFlightMetricsSummary.m]: done

File: 02_helper_functions/AutomatedHallCalibration.m [hunk 3]
```diff
@@ -0,0 +1,129 @@
+function cal_data = AutomatedHallCalibration(sysvector, topics, sensor_sel, cal_opt)
+
+if (strcmp(sensor_sel, 'ALPHA') && topics.sensor_hall.logged)
+    sensor_hall_mag_T = sysvector.sensor_hall_0.mag_T;
+elseif (strcmp(sensor_sel, 'BETA') && topics.sensor_hall_01.logged)   
+    sensor_hall_mag_T = sysvector.sensor_hall_01_0.mag_T;
+else
+    clc;
+    disp('HALL SENS CAL: Logged topics are not sufficient for airspeed calibration.');
```
[the remark this answers is not in the document]
Comment (tstastny) [reply 02_helper_functions/AutomatedHallCalibration.m]: done

File: 08_analysis_tools/CalibrateHallSensor.m (hunk 1 above)
Comment (acfloria) [reply 08_analysis_tools/CalibrateHallSensor.m]: Yes, now it is clearer

**Pull request merged**